### DISCUSSION TITLE
test in Platius model

### DISCUSSION THREAD

**Discussion by trinh-hoang-hiep** [cnguyen10/few_shot_meta_learning#19] [Q&A] [answered]
Hi Cuong,
Thanks for your great work.
I want to ask about the test function in Platius.py what is the meaning of eps_generator in test function

**Comment by cnguyen10** [comment, marked as answer]
Hi Hiep,

The code had been developed a couple of years ago. I then decided to rewrite to put it into one structure specified in `MLBase.py`. Thus, there might remain code adopted from previous implementation.

As what I can recall, the prefix `eps_` denotes "episode" or "task" in meta-learning. Thus, the `eps_generator` is an interative object/function that generates the training and validation split of a task. This had been done before implementing the `EpisodeSampler` class in `EpisodeSampler.py`. In that implementation, a task was generated by simply sampling a number of class labels, e.g. sampling 5 class labels out of 64 classes in mini-ImageNet followed by sampling training and validation split images based on those 5 classes. Such allowed us to gain more control over the tasks/episodes generated rather than a random one. With the implementation of `EpisodeSampler` class, the task generation was integrated into PyTorch dataloader structure, and can run much faster than the old implementation relying on manually sampling tasks.

By the way, if you just want to discuss about the meaning of the code, please open a discussion instead of issue.

**Comment by trinh-hoang-hiep** [comment]
Thanks for your detailed explanation!
a bit off-topic question, I want to apply Platius or vampire to the uncertainty in my image segmentation task. My task is a single task, so there will be no validation sample x_v and training sample x_t, so is there any way I can apply this model to a single task?

**Reply by cnguyen10** [reply]
I do not think that it is feasible since meta-learning is setup under the assumption of task environment. If one wants to use meta-learning, he/she has to formulate the problem into this setting before proceeding further.

For your particular problem, I wonder if you just integrate the modelling uncertainty directly into your segmentation problem without meta-learning. In particular, PLATIPUS and VAMPIRE are the extensions of _Bayes-by-backprop_ method (see the paper "__Weight uncertainty in neural networks__") for meta-learning, while your problem is just for a single task. Hence, I recommend you to read that paper and see if that can be applied to your problem or not.
